### PR TITLE
Made ghost cafe mobs dust upon leaving, allowed ghost cafe bans

### PR DESCRIPTION
**Pull request by Putnam3145**
## About The Pull Request

See title. 

## Why It's Good For The Game

Less exploits are always good. This limits exploit potential massively.

## Changelog
:cl:
tweak: Ghost cafe mobs are super duper attached to the ghost cafe.
/:cl:

### CODE REVIEW

**Review by Ghommie — APPROVED** [review]

**Comment by Ghommie** [comment]
This makes the ghost cafe's hilbertshotel quite useless though.

File: code/datums/elements/dusts_on_leaving_area.dm [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	if(!attached_mobs.len)
+		STOP_PROCESSING(SSprocessing,src)
+
+/datum/element/dusts_on_leaving_area/process()
+	for(var/m in attached_mobs)
+		var/mob/M = m
+		var/area/A = get_area(M)
+		if(!(A.type in area_types))
+			M.dust(force = TRUE)
+			Detach(M)
```
Comment (silicons) [inline code/datums/elements/dusts_on_leaving_area.dm:28]: shouldn't this go before dust as i'm assuming elements auto detach on qdel?

also is it really worth elementing everything when you can just hook ghost cafe area to dust people with the ghost cafe roll or
feels really expensive for a niche  thing =|

Comment (Ghommie) [reply code/datums/elements/dusts_on_leaving_area.dm:28]: While, it should indeed detach automatically on Destoy(), keep in mind that elements are singletons assignable to multiple datums, and are only created if there is no other existing element that meets the criterias (mostly just the type, in certain cases arguments too). TL;DR lightweight components.

Comment (Putnam3145) [reply code/datums/elements/dusts_on_leaving_area.dm:28]: Yeah, this is a "just-in-case", and it won't cause any runtimes or issues AFAIK.

Also, I explicitly wrote this to be non-snowflakey for e.g. weird magical ghost roles in the future.

Also, it's not that expensive.

Also, that wouldn't let ghost cafe mobs enter the hilbert hotel.

**Pull request merged**